synapse/federation/federation_server.py

```diff
@@ -955,6 +949,7 @@ async def on_event_auth(
         self, origin: str, room_id: str, event_id: str
     ) -> Tuple[int, Dict[str, Any]]:
         async with self._server_linearizer.queue((origin, room_id)):
+            await self._event_auth_handler.assert_host_in_room(room_id, origin)
```

Comment:
The two cases above look to be replacing existing checks that the host is in the room. But there doesn't seem to be one here. Is that a bug that this change fixes?